ir/src/constraints/graph.rs

```diff
@@ -166,7 +173,38 @@ impl AlgebraicGraph {
                     lhs.domain(),
                 ))
             }
-            Expression::ListFolding(_) => todo!(),
+            Expression::ListFolding(lf_type) => match lf_type {
+                ListFoldingType::Sum(lc) => {
+                    let list = parse_lc(lc, symbol_table)?;
+                    let mut sum =
+                        self.insert_expr(symbol_table, &list[0], variable_roots, default_domain)?;
+                    for elem in list.iter().skip(1) {
+                        let expr =
+                            self.insert_expr(symbol_table, elem, variable_roots, default_domain)?;
+                        sum = self.insert_bin_op(
+                            &sum,
+                            &expr,
+                            Operation::Add(sum.root_idx(), expr.root_idx()),
+                        )?;
+                    }
+                    Ok(sum)
```

Comment:
It may be possible to create a function which does this for both `Sum` and `Prod` - the only thing which is different between the two is which binary operation we use in the end.

Reply:
Done